src/configuration/rawApiConfiguration.ts

```diff
@@ -0,0 +1,41 @@
+export interface RawApiConfiguration {
+    deployment: DeploymentDefinition;
+}
+
+export interface DeploymentDefinition {
+    codesConverter: [CodeConverter];
+    clientAPICommunication: [ClientApiCommunicationDefinition];
+}
+
+export interface CodeConverter {
+    components: [{ component: [ComponentDefinition] }];
+}
+
+export interface ComponentDefinition {
+    $: { name: string, id: string };
```

Comment:
I'm using this lib ([https://github.com/Leonidas-from-XIV/node-xml2js](url)) to convert xml to js object. $ is the default property to access xml attributes.